code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -284,11 +284,25 @@
 	ratvar_modules = list(
 		/obj/item/clockwork/slab/cyborg/medical,
 		/obj/item/clockwork/weapon/ratvarian_spear)
-	cyborg_base_icon = "medical"
 	moduleselect_icon = "medical"
 	feedback_key = "cyborg_medical"
 	can_be_pushed = FALSE
-	hat_offset = 3
+
+/obj/item/robot_module/medical/be_transformed_to(obj/item/robot_module/old_module)
+	var/mob/living/silicon/robot/R = loc
+	var/borg_icon = input(R, "Select an icon!", "Robot Icon", null) as null|anything in list("Classic", "Droid")

```

Comment:
So that `list("classic", "droid")` could be turned into a single variable for each cyborg type that has an associative list with the names and icon states of each version. That would be a way cleaner implementation.
```
var/list/cyborg_versions = list("Classic" = "medical", "Droid" = "medicaldroid")
... way later in the actual proc ...
if(cyborg_versions.len > 1)
    var/borg_icon = input(R, "Select an icon!", "Robot Icon", null) as null|anything in cyborg_versions```

You should be able to figure out the rest. I'd rather not have people have to code this stuff when they want to add a new cyborg version.